components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServletTest.java

```diff
@@ -0,0 +1,249 @@
+import org.testng.annotations.Test;
+import org.wso2.carbon.core.internal.CarbonCoreDataHolder;
+import org.wso2.carbon.core.util.KeyStoreManager;
+import org.wso2.carbon.identity.application.authentication.framework.CommonAuthenticationHandler;
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkUtils;
+import org.wso2.carbon.identity.core.util.IdentityConfigParser;
+import org.wso2.carbon.identity.core.util.IdentityDatabaseUtil;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.identity.oauth.config.OAuthServerConfiguration;
+import org.wso2.carbon.identity.oauth.tokenprocessor.TokenPersistenceProcessor;
+import org.wso2.carbon.identity.oidc.session.OIDCSessionManager;
+import org.wso2.carbon.identity.oidc.session.util.OIDCSessionManagementUtil;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import java.util.Collections;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.verify;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+@PrepareForTest({OIDCSessionManagementUtil.class, OIDCSessionManager.class, FrameworkUtils.class,
+        IdentityConfigParser.class, OAuthServerConfiguration.class, IdentityTenantUtil.class, KeyStoreManager.class,
+        CarbonCoreDataHolder.class, IdentityDatabaseUtil.class})
+/***
+ * Unit test coverage for OIDCLogoutServlet class
+ */
+public class OIDCLogoutServletTest extends TestOIDCSessionBase {
+
+    @Mock
+    OIDCSessionManager oidcSessionManager;
+
+    @Mock
+    HttpServletRequest request;
+
+    @Mock
+    HttpServletResponse response;
+
+    @Mock
+    CommonAuthenticationHandler commonAuthenticationHandler;
+
+    @Mock
+    HttpSession httpSession;
+
+    @Mock
+    IdentityConfigParser identityConfigParser;
+
+    @Mock
+    OAuthServerConfiguration oAuthServerConfiguration;
+
+    @Mock
+    KeyStoreManager keyStoreManager;
+
+    @Mock
+    TokenPersistenceProcessor tokenPersistenceProcessor;
+
+    private static final String CLIENT_ID_VALUE = "3T9l2uUf8AzNOfmGS9lPEIsdrR8a";
+    private static final String APP_NAME = "myApp";
+    private static final String SECRET = "87n9a540f544777860e44e75f605d435";
+    private static final String USERNAME = "user1";
+
+    private OIDCLogoutServlet logoutServlet;
+
+    @BeforeTest
+    public void setUp() throws Exception {
+
+        logoutServlet = new OIDCLogoutServlet();
+
+        initiateInMemoryH2();
+        createOAuthApp(CLIENT_ID_VALUE, SECRET, USERNAME, APP_NAME, "ACTIVE");
+
+    }
+
+    private static final String OPBROWSER_STATE = "090907ce-eab0-40d2-a46d-acd4bb33f0d0";
```

Comment:
It's better to define all class & instance variables at the beginning of the class.